app/Http/Controllers/Administration/SettingsController.php

```diff
@@ -387,7 +387,7 @@ public function setCSS(Request $request): void
 		$request->validate(['css' => 'present|nullable|string']);
 		$css = $request->get('css') ?? '';
 
-		if (!Storage::disk('dist')->put('user.css', $css)) {
+		if (Storage::disk('dist')->put('user.css', $css) === false) {
```

Comment:
Why that change?

Reply:
Because put() does not return a `bool|string` :
```
 ------ -------------------------------------------------------------------- 
  Line   app/Http/Controllers/Administration/SettingsController.php          
 ------ -------------------------------------------------------------------- 
  390    Only booleans are allowed in a negated boolean, bool|string given.  
 ------ -------------------------------------------------------------------- 
```

Reply:
Sure, but why did it work before? Did Laravel change the return type?

Reply:
File system contract says:
```php
    /**
     * Write the contents of a file.
     *
     * @param  string  $path
     * @param  string|resource  $contents
     * @param  mixed  $options
     * @return bool
     */
    public function put($path, $contents, $options = []);
```
So I guess this is a problem of the phpstan/larastan...

Reply:
Correction, it is once again a stupid problem of Laravel not following their own contract...
```php
    /**
     * Write the contents of a file.
     *
     * @param  string  $path
     * @param  \Psr\Http\Message\StreamInterface|\Illuminate\Http\File|\Illuminate\Http\UploadedFile|string|resource  $contents
     * @param  mixed  $options
     * @return string|bool
     */
    public function put($path, $contents, $options = [])
    {
        $options = is_string($options)
                     ? ['visibility' => $options]
                     : (array) $options;

        // If the given contents is actually a file or uploaded file instance than we will
        // automatically store the file using a stream. This provides a convenient path
        // for the developer to store streams without managing them manually in code.
        if ($contents instanceof File ||
            $contents instanceof UploadedFile) {
            return $this->putFile($path, $contents, $options);
        }

        try {
            if ($contents instanceof StreamInterface) {
                $this->driver->writeStream($path, $contents->detach(), $options);

                return true;
            }

            is_resource($contents)
                ? $this->driver->writeStream($path, $contents, $options)
                : $this->driver->write($path, $contents, $options);
        } catch (UnableToWriteFile|UnableToSetVisibility $e) {
            throw_if($this->throwsExceptions(), $e);

            return false;
        }

        return true;
    }
```

Reply:
I see great opportunities for another hopeless PR towards Laravel. :rofl: